### PR TITLE
Proxy-supplied peerAddress

### LINKED ISSUE #31: Extensible Peer Address from Proxies
Come up with a good way to get the peerAddress from a proxy, for example, Nginx uses the X-Forwarded-For header.

### PR DESCRIPTION
**Pull request by czechboy0**
Fixes #31. cc @loganwright

### CODE REVIEW

**Comment by tannernelson** [comment]
This is awesome. 

Would there be any way to get access to the underlying Stream peer address? Users will always have access to the "forwarded" peer addresses from the headers, but it looks like there's no way to access the underlying stream peer address.

**Comment by czechboy0** [comment]
@tannernelson That's a good point, right now we're blindly trusting the headers. Would this look okay?

```struct
struct PeerAddress {
    let xForwardedFor: String?
    let forwarded: String?
    let stream: String
}
```

**Comment by tannernelson** [comment]
Looks perfect.

File: Sources/HTTP/Parser/HTTPParser.swift [hunk 1]
```diff
@@ -96,36 +96,13 @@ public final class Parser<MessageType: Message>: TransferParser {
...
-            return xForwardedFor
-        }
-        
-        // 3.
-        return stream.peerAddress
+    private func parsePeerAddress(headers: [HeaderKey: String]) -> PeerAddress {
+        let forwarded = headers["Forwarded"]
+        let xForwardedFor = headers["X-Forwarded-For"]
+        let streamAddress = stream.peerAddress
+        return PeerAddress(forwarded: forwarded,
```
Comment (tannernelson) [inline Sources/HTTP/Parser/HTTPParser.swift]: No Xcode formatting please :)

```swift
return PeerAddress(
    forwarded: asdf,
    ...
)
```

**Comment by tannernelson** [comment]
Something strange with the tests...

**Comment by czechboy0** [comment]
Formatting is fixed, the issue with tests is with CI services: Circle doesn't seem to be configured (did you recently enable it?), so it's failing. Travis Mac builds have been down for a day. I tested locally, all works (test yourself if you want).

**Pull request merged**